src/topp/MzTabExporter.cpp

```diff
@@ -437,38 +514,87 @@ namespace OpenMS
           const std::vector<ProteinIdentification::ProteinGroup> indist_groups = it->getIndistinguishableProteins();
           const std::vector<ProteinHit> protein_hits = it->getHits();
 
+          // pre-analyze data for occuring meta values at protein hit level
+          // these are used to build optional columns containing the meta values in internal data structures
+          set<String> protein_hit_user_value_keys;
+          for (Size i = 0; i < protein_hits.size(); ++i)
```

Comment:
For Feature it should work but for the lower level PeptideIdentification I think either needs to define custom iterators or use  something like http://www.boost.org/doc/libs/1_58_0/libs/iterator/doc/iterator_adaptor.html to e.g. generate a "deep iterator" that iterates over all meta values of the underlying PeptideIdentification objects of each Feature. But maybe there are other / better ways. C++ guru to the rescue?